app/classes/box.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# rectangle on the surface of the earth, with borders: north, south, east, west
+# used mostly (exclusively?) by model scopes
+class Box
+  attr :north, :south, :east, :west
+
+  def initialize(north: nil, south: nil, east: nil, west: nil)
+    @north = north
+    @south = south
+    @east = east
+    @west = west
+  end
+
+  def valid?
+    args_in_bounds? &&
+      south <= north &&
+      ((west <= east) || straddles_180_deg?)
+  end
+
+  def straddles_180_deg?
+    west > east && (west >= 0 && east <= 0)
```

Comment:
Seems to me it would be possible for a box, at least theoretically, to go from, say, -10° in the west, all the way around, cross the date line and end at -170°.  In other words, there should be no restrictions on east and west... well, except that they should all be valid degrees: north <= 90, south >= -90, -180 <= east <= 180, -180 <= west <= 180, and north >= south.  And maybe north != south and east != west?

Reply:
Thank you. You're right. 
I'll fix it.